extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -2122,13 +2123,33 @@ private static StringTuple makeStringTuple(
     return new StringTuple(array);
   }
 
+  private static DimensionSchema getDimensionSchema(
+      final String outputColumnName,
+      @Nullable final ColumnType queryType,
+      QueryContext context,
+      @Nullable Map<String, DimensionSchema> dimensionToSchemaMap
+  )
+  {
+    if (dimensionToSchemaMap != null && dimensionToSchemaMap.containsKey(outputColumnName)) {
+      return dimensionToSchemaMap.get(outputColumnName);
+    }
+    // For aggregators moved to dimensions, we won't have an entry in the map. For those cases, use the default config.
```

Comment:
this also happens for like regular ingestions which are not compaction, right? comment makes it seem only like aggs

Reply:
Yes, thanks for pointing out. Updated the comment.